src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -82,23 +90,33 @@ class MongoDbStore implements StoreInterface
      *
      * @see https://docs.mongodb.com/manual/applications/replication/
      */
-    public function __construct(Client $mongo, array $options, float $initialTtl = 300.0)
+    public function __construct($mongo, array $options = [], float $initialTtl = 300.0)
     {
-        $this->mongo = $mongo;
-
-        $this->initialTtl = $initialTtl;
-
         $this->options = array_merge([
-            'collection' => 'lock',
             'gcProbablity' => 0.001,
+            'collection' => 'lock',
+            'uriOptions' => [],
+            'driverOptions' => [],
         ], $options);
 
-        if (!isset($this->options['database'])) {
-            throw new InvalidArgumentException('You must provide the "database" option for MongoDBStore');
+        $this->initialTtl = $initialTtl;
+
+        if ($mongo instanceof Collection) {
+            $this->collection = $mongo;
+        } elseif ($mongo instanceof Client) {
+            $this->client = $mongo;
+        } elseif (is_string($mongo)) {
+            $this->dsn = $mongo;
+        } else {
+            throw new InvalidArgumentException(sprintf('%s() requires %s or %s or DSN as first argument, "%s" given.', __METHOD__, Collection::class, Client::class, \is_object($mongo) ? \get_class($mongo) : \gettype($mongo)));
+        }
+
+        if (null === $this->collection && !isset($this->options['database'])) {
+            throw new InvalidArgumentException(sprintf('%s() requires the "database" option when constructing with a %s', __METHOD__, \is_object($mongo) ? \get_class($mongo) : \gettype($mongo)));
         }
 
         if ($this->options['gcProbablity'] < 0.0 || $this->options['gcProbablity'] > 1.0) {
-            throw new InvalidArgumentException(sprintf('"%s" gcProbablity must be a float from 0.0 to 1.0, "%f" given.', __METHOD__, $this->options['gcProbablity']));
+            throw new InvalidArgumentException(sprintf('%s() gcProbablity must be a float from 0.0 to 1.0, "%f" given.', __METHOD__, $this->options['gcProbablity']));
```

Comment:
remove `"` in `"%f" given` too (for consistency)

Reply:
updated